src/xcode/ENA/ENA/Source/Services/CoronaTestService/CoronaTestService.swift

```diff
@@ -468,6 +470,31 @@ class CoronaTestService {
 		store.isSubmissionConsentGiven = false
 	}
 
+	func setUniqueCertificateIdentifier(set uniqueCertificateIdentifier: String, with testCertificateRequest: TestCertificateRequest) {
+		switch testCertificateRequest.coronaTestType {
+		case .pcr:
+			if self.pcrTest?.registrationToken == testCertificateRequest.registrationToken {
+				pcrTest?.uniqueCertificateIdentifier = uniqueCertificateIdentifier
+			}
+		case .antigen:
+			if self.antigenTest?.registrationToken == testCertificateRequest.registrationToken {
+				self.antigenTest?.uniqueCertificateIdentifier = uniqueCertificateIdentifier
+			}
+		}
+	}
+	
+	func getHealthCertificateTuple(for uniqueCertificateIdentifier: String) -> HealthCertificateTuple? {
+		var healthTuple: HealthCertificateTuple?
+		self.healthCertificateService.healthCertifiedPersons.value.forEach { healthCertifiedPerson in
+			healthCertifiedPerson.healthCertificates.forEach { healthCertificate in
+				if healthCertificate.uniqueCertificateIdentifier == uniqueCertificateIdentifier {
+					healthTuple = HealthCertificateTuple(certificate: healthCertificate, certifiedPerson: healthCertifiedPerson)
+				}
+			}
+		}
+		return healthTuple
+	}
```

Comment:
I think if you just need it here you can just actually use a tuple. And we don't need the `get` prefix in Swift 😉

```suggestion
	func healthCertificateTuple(for uniqueCertificateIdentifier: String) -> (certificate: HealthCertificate, certifiedPerson: HealthCertifiedPerson)? {
		var healthTuple: (certificate: HealthCertificate, certifiedPerson: HealthCertifiedPerson)?
		self.healthCertificateService.healthCertifiedPersons.value.forEach { healthCertifiedPerson in
			healthCertifiedPerson.healthCertificates.forEach { healthCertificate in
				if healthCertificate.uniqueCertificateIdentifier == uniqueCertificateIdentifier {
					healthTuple = (certificate: healthCertificate, certifiedPerson: healthCertifiedPerson)
				}
			}
		}
		return healthTuple
	}
```